.github/workflows/tox.yml

```diff
@@ -340,3 +340,34 @@ jobs:
       #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       #   # https://github.com/sourcegraph/codenotify/issues/19
       #   continue-on-error: true
+
+  notify:
+    if: always() && (needs.test.result == 'failure' || needs.check.result == 'failure') && github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs:
+      - test
+      - check
```

Comment:
There is no reason to create a different job for notify, just use `check` one and add the extra steps to it. Each job adds computing costs and time.